### PR TITLE
Added release notes for new operator release and updated the document…

### PR DESCRIPTION
**Pull request by Nidhi251289**
…ation for startupProbe and terminationGracePeriodSeconds

### CODE REVIEW

File: content/en/docs/releasenotes/deployment/mendix-for-private-cloud.md [hunk 1]
```diff
@@ -13,6 +13,18 @@ For information on the current status of deployment to Mendix for Private Cloud
...
+### July  ???, 2022
+
+#### Mendix Operator v2.6.0{#2.6.0}
+
+* We improved the default health check configuration by allowing the use of Kubernetes startup probes. This feature improves the reliability of environments that need more time to initialize, for example when executing database migration startup microflows.
+* We now allow you to customize the Kubernetes `terminationGracePeriodSeconds` attribute for apps, allowing an app to have more time to perform a clean shutdown — for example to close database connections and complete microflows and scheduled events that are already running.
+* We fixed a build error which happened when an MDA included a data snapshot.
+* We resolved an issue where the sidecar container didn’t process the shutdown signal, even when the app container was stopped (this meant that stopping an app took 30 seconds)
+* We have disabled the `enableServiceLinks` Kubernetes feature — this prevents app pods from receiving a list of all services running in a namespace through environment variables.
+* When connecting to the Development Portal, the Mendix Gateway Agent will now trust CAs specified through Custom TLS Trust
```
Comment (MarkvanMents) [inline content/en/docs/releasenotes/deployment/mendix-for-private-cloud.md]: Can we put a link to where we mention "Custom TLS Trust" I can't find that phrase in our documentation.

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/_index.md [hunk 2]
```diff
@@ -1162,7 +1172,49 @@ The following fields can be configured:
...
+startupProbe:
+  httpGet:
+    path: /
+    port: mendix-app
+    scheme: HTTP
+  failureThreshold: 30
+  periodSeconds: 10
+```
+
+In this example, the application will have a maximum of 5 minutes (30 * 10 = 300s) to finish its startup. Once the startup probe has succeeded once, the liveness probe takes over to provide a fast response to container deadlocks. If the startup probe never succeeds, the container is killed after 300s and subject to the pod's restartPolicy.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/_index.md]: I can't find anywhere that the `restartPolicy` is mentioned in our documentation. Can we add it to our documentation or link the the K8s documentation which covers it?

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/_index.md [hunk 3]
```diff
@@ -1162,7 +1172,49 @@ The following fields can be configured:
...
 {{% /alert %}}
 
-#### 5.3.2 Customize Container Resources: Memory and CPU
+#### 5.3.2 Customize startupProbes for slow starting applications
+
+If you want to wait before executing a liveness probe you should use `initialDelaySeconds` or a startupProbe.
+
+A startupProbe should be used when the application in your container could take a significant amount of time to reach its normal operating state. Applications that would crash or throw an error if they handled a liveness or readiness probe during startup need to be protected by a startupProbe. This ensures the container doesn't continually restart due to failing health checks before it has finished launching. Using a startupProbe is much better than increasing `initialDelaySeconds` on readiness or liveness probes. StartupProbes defer the execution of liveness and readiness probes until a container indicates it is able to handle them because Kubernetes doesn't direct the other probe types to the container if it has a startupProbe that hasn't yet succeeded.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/_index.md]: Should I be calling them "startupProbe" as in the setting, or would it be better to use natural language and say "startup probe" except when I am specifically talking about the YAML key?

**Review by MarkvanMents — CHANGES_REQUESTED** [review]
Hi Nidhi (@Nidhi251289)
I have proofread the documents - let me know if I've misunderstood anything.
I have also added a couple of places where I think we could point to further documentation to explain things.
Let me know if you have any questions.
Mark

PS - there were a lot of files called `.DS_Store` in the Pull Request. Not sure how they got there, but they aren't needed for our site so I've deleted them. Can you check that you aren't adding them back in if you make any more commits. Thanks.

File: content/en/docs/releasenotes/deployment/mendix-for-private-cloud.md (hunk 1 above)
Comment (Nidhi251289) [reply content/en/docs/releasenotes/deployment/mendix-for-private-cloud.md]: https://docs.mendix.com/developerportal/deploy/private-cloud-cluster/#custom-tls is the link

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/_index.md (hunk 2 above)
Comment (Nidhi251289) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/_index.md]: ok. https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy
We can add above link

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/_index.md (hunk 3 above)
Comment (Nidhi251289) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/_index.md]: if we are referring to yaml, then startupProbe
else, startup probe

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/_index.md [hunk 4]
```diff
@@ -1247,11 +1299,15 @@ spec:
...
+  runtimeStartupProbe:
+    failureThreshold: 30
+    periodSeconds: 10
+  runtimeTerminationGracePeriodSeconds: 300    
 ```
 
 The following fields can be configured:
 
-* `Liveness` and `readiness` probes  – these are used for all Mendix app deployments in the namespace. Therefore, any changes made in the Deployments will be discarded and overwritten with values from `OperatorConfiguration` resource
+* `Liveness`, `readiness`, `startupProbe`, and `terminationGracePeriodSeconds` – these are used for all Mendix app deployments in the namespace — any changes made in the Deployments will be discarded and overwritten with values from the `OperatorConfiguration` resource
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/_index.md:1309]: Two questions:

1. Shouldn't this be `runtimeLivenessProbe`, `runtimeReadinessProbe`, `runtimeStartupProbe`, and `runtimeTerminationGracePeriodSeconds` these are the values in the YAML
2. Does it make sense for the `runtimeStartupProbe` and `runtimeTerminationGracePeriodSeconds` in the deployment to be overwritten by the `OperatorConfiguration`? Surely, you want to be able to set the `startupProbe` and `terminationGracePeriodSeconds` for each app in the namespace, as indicated in the new documentation we have just added? Or are these values optional and the customer can use them if they want to set these values at the namespace level?

**Review by MarkvanMents — CHANGES_REQUESTED** [review]
Hi Nidhi,
I've made the changes you suggest, but have one more question about the Operator Configuration Manifest.
Then I think we are done if you are happy with my changes.

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/_index.md (hunk 4 above)
Comment (Nidhi251289) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/_index.md:1309]: In general, the above terms are known as Liveness`, `readiness`, `startupProbe`, and `terminationGracePeriodSeconds. The runtime term is used specifically in our CRDS.

The thing is normally the values are already set on namespace level. The above values are default values which will come into effect if the customer dont configure anything on the operator level.

Comment (MarkvanMents) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/_index.md:1309]: OK - I'll leave it for now. We can pick it up if customers have any questions later.
Thanks for the reply.

**Review by MarkvanMents — APPROVED** [review]

**Comment by MarkvanMents** [comment]
Released 12 July.

**Pull request merged**